torchvision/io/video.py

```diff
@@ -179,19 +183,37 @@ def read_video(filename, start_pts=0, end_pts=None):
         raise ValueError("end_pts should be larger than start_pts, got "
                          "start_pts={} and end_pts={}".format(start_pts, end_pts))
 
+    if pts_unit == 'pts':
+        warnings.warn("The pts_unit 'pts' gives wrong results and will be removed in a " +
+                      "follow-up version. Please use pts_unit 'sec'.")
+
     container = av.open(filename, metadata_errors='ignore')
     info = {}
 
     video_frames = []
     if container.streams.video:
-        video_frames = _read_from_stream(container, start_pts, end_pts,
-                                         container.streams.video[0], {'video': 0})
-        info["video_fps"] = float(container.streams.video[0].average_rate)
+        video_start_pts = start_pts
+        video_end_pts = end_pts
+        video_stream = container.streams.video[0]
+        if pts_unit == 'sec':
+            video_start_pts = math.floor(start_pts * (1 / video_stream.time_base))
```

Comment:
will math.floor cast it to int?

Reply:
No, for python2 it is returning answer as float and for python3 it is returning int.

Reply:
Hi, I have converted ouput of math.floor to int for handling the python2 problem.